src/parsers/manifestjson.js

```diff
@@ -22,16 +24,33 @@ export default class ManifestJSONParser {
 
     try {
       this.parsedJSON = JSON.parse(jsonString);
-    } catch (error) {
-      var errorData = {
-        code: 'MANIFEST_JSON_INVALID',
-        message: 'Invalid JSON in manifest file.',
-        file: 'manifest.json',
-        description: error,
-      };
-      this.collector.addError(errorData);
-      this.isValid = false;
-      return;
+    } catch (originalError) {
+      // First we'll try to remove comments with esprima;
+      // WebExtension manifests can contain comments, so we'll strip
+      // them out and see if we can parse the JSON.
+      // If not it's just garbage JSON and we error.
+      //
+      // Originally from https://github.com/abarreir/crx2ff/blob/d2b882056f902d751ad05e329efda7eddcb9d268/libs/ext-converter.js#L19-L37
+      var manifestString = `var o = ${jsonString}`;
+      try {
+        var tokens = esprima.tokenize(manifestString).slice(3);
+        var cleanedJSONString = tokens.reduce((json, token) => {
+          return `${json}${token.value}`;
+        }, '');
```

Comment:
Hmm, okay. So I will document that in the code, but these seem like quite extreme edge cases. I can't imagine we'll have users with these kinds of characters in their manifests.

I'll see about catching those very specific errors and if not I'll at least add those links and some extra comments to the code. Thanks!